qiskit_experiments/characterization/t2ramsey.py

```diff
@@ -99,3 +102,203 @@ def circuits(self, backend: Optional[Backend] = None) -> List[QuantumCircuit]:
+
+    The first (second) circuit measures :math:`\sigma_Y` (:math:`\sigma_X`) expectation value,
+    so this experiment draws the dynamics of the Bloch vector as if drawing a Lissajous figure.
+
+    Given the control electronics tracks the frame of qubit at the reference frequency
+    which is slightly differ from the true qubit frequency by :math:`\Delta\omega`,
+    and the IQ mixer skew can be ignored, we can describe the dynamics of
+    two circuits as follows.
+
+    The Hamiltonian during the ``Delay`` instruction can be written as
+    :math:`H^R = - \frac{1}{2} \Delta\omega` in the rotating frame,
+    and the propagator will be :math:`U(\tau) = \exp(-iH^R\tau)` where :math:`\tau` is the
+    duration of the delay. By scanning this duration, we can get
+
+    .. math::
+
+        {\cal E}_x(\tau)
+            = {\rm Re} {\rm Tr}\left( \sigma_Y U \rho U^\dagger \right)
+            &= - \cos(\Delta\omega\tau) = \sin(\Delta\omega\tau - \frac{\pi}{2}), \\
+        {\cal E}_y(\tau)
+            = {\rm Re} {\rm Tr}\left( \sigma_X U \rho U^\dagger \right)
+            &= \sin(\Delta\omega\tau),
+
+    where :math:`\rho = | L \rangle` prepared by the :math:`\sqrt{\rm X}` gate in the beginning.
+
+    Note that phase difference of these two outcomes :math:`{\cal E}_x, {\cal E}_y` depends on
+    the frequency offset :math:`\Delta\omega`, which is also sensitive
+    to the sign of it, in contrast to the standard Ramsey experiment which
+    usually consists only of the first circuit,
+    i.e. :math:`\cos(-\Delta\omega\tau) = \cos(\Delta\omega\tau)`.
+    """
+
+    __analysis_class__ = RamseyXYAnalysis
+
+    def __init__(
+            self,
+            qubit: int,
+            delays: Union[Iterable[float], np.ndarray],
+            unit: str = "s",
+            offset_frequency: float = 0.,
+    ):
+        """Initialize new experiment.
+
+        Args:
+            qubit: The qubit under test.
+            delays: Delay times of the experiments.
+            unit: Time unit of `delays`. One of 's', 'ms', 'us', 'ns', 'ps', 'dt'.
+            offset_frequency: Frequency offset for this experiment.
+
+        .. warning::
+
+            ``offset_frequency`` doesn't really change the qubit frequency for now,
+            because of the missing oscillator frequency upgrade pass in the circuit execution.
+        """
+        super().__init__([qubit])
+
+        default_options = self.experiment_options
+        default_options.delays = delays
+        default_options.unit = unit
+        default_options.offset_frequency = offset_frequency
+
+        self.set_experiment_options(**default_options.__dict__)
+
+    @classmethod
+    def _default_experiment_options(cls) -> Options:
+        return Options(
+            delays=None,
+            unit="s",
+            offset_frequency=0.,
+            calibrations={
+                "sx": None,
+                "measure": None,
+            },
```

Comment:
Yes, I think integration with transpiler is the final goal, though we need to wait until, at least, next terra release. This is kind of temporary procedure (we can move this code to the experiment base class for now, if you feel this is right direction). Actually the post process transpiler method is kind of pass. 

I don't think gate-pulse name map is necessary. For example, inst map always saves entry with name of gates. This should be managed by the `Calibrations` class, i.e. this should be able to provide inst_map like thing.

Reply:
Regarding frequency, I think this can go to either transpiler or assembler. We can add preamble property to `QuantumCircuit` and generate QASM3 caldef field based on that information. Seems like this is future extension (meaning we cannot run any fine frequency tuning for now...)

Reply:
Adding a gate-schedule map to calibrations is fine. Why do we need to wait for the next release of terra? We could have a light-weight transpiler pass in calibrations that takes care of this. See #180 .